### PR TITLE
refactor: SfDropdown changing api

### LINKED ISSUE #1333: SfDropdown API Changes
Currently to use`SfDropdown` you have to put its trigger in a relatively positioned wrapper: 
```HTML
  <div style="position: relative; display: inline-block;">
        <SfButton @click="isOpen = true">Choose your action</SfButton>
        <SfDropdown :class="customClass" :is-open="isOpen" :persistent="persistent" :title="title" @click:close="isOpen = false">
          <SfList>
            <SfListItem v-for="(action, key) in actionList" :key="key">
              <SfButton class="sf-button--full-width sf-button--underlined color-primary" @click="isOpen = !isOpen">{{ action }}</SfButton>
            </SfListItem>
          </SfList>
        </SfDropdown>
      </div>
```

As a better DX I suggest following API

```html
<SfDropdown v-slot="toggle">
   <SfButton>Choose your action</SfButton>
    <template #content>
        <SfList>
            <SfListItem v-for="(action, key) in actionList" :key="key">
              <SfButton class="sf-button--full-width sf-button--underlined color-primary" @click="isOpen = !isOpen">{{ action }} </SfButton>
           </SfListItem>
        </SfList>
    </template>
</SfDropdown>
```
In the default slot we're keeping dropdown toggle fn with `toggle` function as slot scope.
Toggle function should behave as following
- `toggle()` changes state to the opposite
- `toggle(true)` - changes open state to true
- `toggle(false)` - changes open state to false

In `content` slot by ddefault we have an empty box which can be filled with additional content like in the example

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue

Closes #1333 

# Scope of work
Changes in api:
add SfButton openning items inside component and add toggle funcion. 

# Screenshots of visual changes

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed

### CODE REVIEW

**Comment by AdamPawlinski** [comment]
API changed. There are two methods open and close emitting events. Toggle function didn't work for me.

File: packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue [hunk 1]
```diff
@@ -1,6 +1,11 @@
 <template>
   <div class="sf-dropdown">
     <SfOverlay :visible="isOpen" class="sf-dropdown__overlay" />
+    <slot>
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue:4]: Imho if we have a default opener it should be a default slot. default slot should be for a content, and for opener (as its optional) we should have it being a named slot like "opener"

**Comment by filrak** [comment]
information about changed api is missing in  upgrade Notes

File: packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue (hunk 1 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue:4]: Ok, I'll fix that.

File: packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue [hunk 2]
```diff
@@ -77,15 +82,18 @@ export default {
...
   methods: {
+    open() {
+      this.$emit("click:open");
+    },
     close() {
       this.$emit("click:close");
     },
     checkPersistence() {
-      if (!this.persistent) this.close();
+      if (!this.persistent) return this.$emit("click:close");
```
Comment (Michal-Dziedzinski) [inline packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue:92]: Why did you decide to replace close method?

Comment (AdamPawlinski) [reply packages/vue/src/components/molecules/SfDropdown/SfDropdown.vue]: In fact, I don't know :)  It should stay like it was.

**Review by filrak — APPROVED** [review]

**Pull request merged**